### PR TITLE
doc: add plugins contribution doc

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Add doc which describes how the SDK plugins work and provide the required info for who would like to create your own plugins following the same standards as for its maintainers. 

Preview: https://deploy-preview-4732--operator-sdk.netlify.app/docs/contribution-guidelines/plugins/

### CODE REVIEW

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+Also, see the topic [Language-based Plugins][kb-language-plugins] to understand know how it works.
+
+### Common scaffolds
+
+| File/Directory | Description | 
+| ------ | ----- |
+|  Dockerfile |  Defines the operator(manager) image |  
+|  Makefile |  Provide the helpers and options for the users. (e.g. `make bundle` which generate/update the OLM [bundle][bundle] manifests ) |  
+|  PROJECT |  Project configuration. Store the data used to do the scaffolds. For further information see [Project Config][kb-project] | 
+|  bundle.Dockerfile |  Docker image which is used to provide the helpers to integate the project with OLM. (e.g. [operator-sdk run ./bundle][sdk-cli-run-bundle]) |
```
Comment (jmrodri) [inline website/content/en/docs/contribution-guidelines/plugins.md:30]:
```suggestion
|  bundle.Dockerfile | Docker image which is used to provide the helpers to integrate the project with OLM. (e.g. [operator-sdk run ./bundle][sdk-cli-run-bundle]) | 
```

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+|  bundle.Dockerfile |  Docker image which is used to provide the helpers to integate the project with OLM. (e.g. [operator-sdk run ./bundle][sdk-cli-run-bundle]) | 
+|  config/ |  Directory which has all [kustomize's][kustomize] manifest to configure and test the project | 
+
+You can check the [Project Layout][project-layout] to understand better the files and directories which are scaffold by SDK CLI which are common or not for each Language-type. 
+
+## Custom Plugins
+
+By default, some plugins are used by SDK to add the required changes in the scaffolds to provide its features which currently are: 
+
+- [manifests.sdk.operatorframework.io][plugin-manifest]: resposanble to perform all required changes in the scaffolds and provide the helpers to allow the projects to be integrated with OLM.
```
Comment (jmrodri) [inline website/content/en/docs/contribution-guidelines/plugins.md:39]:
```suggestion
- [manifests.sdk.operatorframework.io][plugin-manifest]: responsible to perform all required changes in the scaffolds and provide the helpers to allow the projects to be integrated with OLM. 
```

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+|  config/ |  Directory which has all [kustomize's][kustomize] manifest to configure and test the project | 
+
+You can check the [Project Layout][project-layout] to understand better the files and directories which are scaffold by SDK CLI which are common or not for each Language-type. 
+
+## Custom Plugins
+
+By default, some plugins are used by SDK to add the required changes in the scaffolds to provide its features which currently are: 
+
+- [manifests.sdk.operatorframework.io][plugin-manifest]: resposanble to perform all required changes in the scaffolds and provide the helpers to allow the projects to be integrated with OLM. 
+- [scorecard.sdk.operatorframework.io][plugin-scorecard]: resposanble to scaffold all files and helpers required to provide the [Scorecard][scorecard] feature
```
Comment (jmrodri) [inline website/content/en/docs/contribution-guidelines/plugins.md:40]:
```suggestion
- [scorecard.sdk.operatorframework.io][plugin-scorecard]: responsible to scaffold all files and helpers required to provide the [Scorecard][scorecard] feature
```

**Review by jmrodri — CHANGES_REQUESTED** [review]
a few changes

**Comment by camilamacedo86** [comment]
All suggestions are addressed. Thank you.

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+[project-layout]: ../../docs/overview/project-layout
+[plugin-manifest]: https://github.com/operator-framework/operator-sdk/tree/master/internal/plugins/manifests/v2
+[plugin-scorecard]: https://github.com/operator-framework/operator-sdk/tree/master/internal/plugins/scorecard/v2
+[kubebuilder-declarative-pattern]: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern
+[kubebuilder-declarative-pattern-example]: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/controllers/firstmate_controller.go
+[default-scaffold]: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/controllers/admiral_controller.go
+[kb-issue]: https://github.com/kubernetes-sigs/kubebuilder/issues/1378
+[create-your-own-plugins]: https://deploy-preview-2115--kubebuilder.netlify.app/plugins/creating-plugins.html
+[scorecard]: ../../docs/advanced-topics/scorecard
+[kubebuilder]: https://github.com/kubernetes-sigs/kubebuilder
```
Comment (camilamacedo86) [inline website/content/en/docs/contribution-guidelines/plugins.md:78]: The `https://deploy-preview-2115--kubebuilder` requires to be replaced when the PR https://github.com/kubernetes-sigs/kubebuilder/pull/2115 gt merged.

**Review by jmrodri — APPROVED** [review]

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+### Optional/custom plugins
+
+Users also can use custom plugins when they execute the SDK CLI sub-commands as a helper for its options which would like:
+
+```sh
+operator-sdk create api --plugins="go/v3,declarative"
+```
+
+The above example will scaffold a custom code in the controllers after when an api is created to allow the users develop solutions using the [kubebuilder declarative pattern][kubebuilder-declarative-pattern]. (e.g [default scaffold][default-scaffold] versus [example][kubebuilder-declarative-pattern-example]).
```
Comment (marc-obrien) [inline website/content/en/docs/contribution-guidelines/plugins.md]: The above example will scaffold a custom code in the controllers after an api is created to allow the users to develop solutions using the [kubebuilder declarative pattern][kubebuilder-declarative-pattern]. (e.g [default scaffold][default-scaffold] versus [example][kubebuilder-declarative-pattern-example]).

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Note that custom plugins which are called via the init sub-command, will be added in the `layout` field of the PROJECT file and will work such as global plugins. Then, it means that for any sub-command executed; they will also be called.
```
Comment (marc-obrien) [inline website/content/en/docs/contribution-guidelines/plugins.md:54]: Note that custom plugins are called via the init sub-command to work as global plugins, and will be added in the `layout` field of the PROJECT file. Any sub-command executed will then also be called.

**Review by marc-obrien — CHANGES_REQUESTED** [review]
Please find comments to address

File: website/content/en/docs/contribution-guidelines/plugins.md (hunk 6 above)
Comment (camilamacedo86) [reply website/content/en/docs/contribution-guidelines/plugins.md]: done.

File: website/content/en/docs/contribution-guidelines/plugins.md (hunk 5 above)
Comment (camilamacedo86) [reply website/content/en/docs/contribution-guidelines/plugins.md]: done.

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+| File/Directory | Description | 
+| ------ | ----- |
+|  Dockerfile |  Defines the operator(manager) image |  
+|  Makefile |  Provides the helpers and options for the users. (e.g. `make bundle` which generate/update the OLM [bundle][bundle] manifests ) |  
+|  PROJECT |  Project configuration. Stores the data used to do the scaffolds. For further information see [Project Config][kb-project] | 
+|  bundle.Dockerfile | Docker image which is used to provide the helpers to integrate the project with OLM. (e.g. [operator-sdk run ./bundle][sdk-cli-run-bundle]) | 
+|  config/ |  Directory which has all [kustomize's][kustomize] manifest to configure and test the project | 
+
+You can check the [Project Layout][project-layout] to better understand the files and directories scaffold by SDK CLI, which are common or not for each Language-type.
```
Comment (varshaprasad96) [inline website/content/en/docs/contribution-guidelines/plugins.md:35]: Nit: 
```suggestion
You can check the [Project Layout][project-layout] to better understand the files and directories scaffolded by SDK CLI, which may be common for each language-type. 
```

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+## Custom Plugins
+
+By default, some plugins are used by SDK to add the required changes in the scaffolds to provide its features which currently are: 
+
+- [manifests.sdk.operatorframework.io][plugin-manifest]: perform the required scaffolds to provide the helpers to allow the projects to be integrated with OLM. 
+- [scorecard.sdk.operatorframework.io][plugin-scorecard]: perform the required scaffolds to provide the [Scorecard][scorecard] feature.
+
+### Optional/custom plugins
+
+Users also can use custom plugins when they execute the SDK CLI sub-commands as a helper for its options which would like:
```
Comment (varshaprasad96) [inline website/content/en/docs/contribution-guidelines/plugins.md:46]: Nit:
```suggestion
Users can also use custom plugins when they execute the SDK CLI sub-commands as a helper for its options which could be like:
```

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+operator-sdk create api --plugins="go/v3,declarative"
+```
+
+The above example will scaffold a custom code in the controllers after an API is created to allow the users to develop solutions using the [kubebuilder declarative pattern][kubebuilder-declarative-pattern]. (e.g [default scaffold][default-scaffold] versus [example][kubebuilder-declarative-pattern-example]).
+
+Note that custom plugins are called via the init sub-command to work as global plugins, and will be added in the `layout` field of the PROJECT file. Any sub-command executed will then also be called.
+
+## Plugins Vision
+
+Maintainers and contributors are able to create their own plugin(s) using the same standards and approaches. Theoretically, that could be an option to allow these solutions to be compatible and useful with SDK CLI in the future or at least could result in lower effort to add them on this project, if that is its ultimate motivation. Currently, SDK cannot discover and use plugins that are not ship or build with its binary. However, it has been discussed in the issue [kubebuilder/issues/1378][kb-issue].
```
Comment (varshaprasad96) [inline website/content/en/docs/contribution-guidelines/plugins.md:58]:
```suggestion
Maintainers and contributors are able to create their own plugin(s) using the same standards and approaches. Theoretically, that could be an option to allow these solutions to be compatible and useful with SDK CLI in the future or at least could result in lower effort to add them on this project, if that is its ultimate motivation. Currently, SDK cannot discover and use plugins that are not shipped or built with its binary. However, it has been discussed in the issue [kubebuilder/issues/1378][kb-issue].
```

**Review by varshaprasad96 — APPROVED** [review]
few nits and lgtm

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+|  Makefile |  Provides the helpers and options for the users. (e.g. `make bundle` which generate/update the OLM [bundle][bundle] manifests ) |  
+|  PROJECT |  Project configuration. Stores the data used to do the scaffolds. For further information see [Project Config][kb-project] | 
+|  bundle.Dockerfile | Docker image which is used to provide the helpers to integrate the project with OLM. (e.g. [operator-sdk run ./bundle][sdk-cli-run-bundle]) | 
+|  config/ |  Directory which has all [kustomize's][kustomize] manifest to configure and test the project | 
+
+You can check the [Project Layout][project-layout] to better understand the files and directories scaffolded by SDK CLI, which may be common for each language-type.
+
+## Custom Plugins
+
+By default, some plugins are used by SDK to add the required changes in the scaffolds to provide its features which currently are:
```
Comment (marc-obrien) [inline website/content/en/docs/contribution-guidelines/plugins.md:39]: By default plugins are used by the Operator SDK to provide the following features:

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+## Custom Plugins
+
+By default, some plugins are used by SDK to add the required changes in the scaffolds to provide its features which currently are: 
+
+- [manifests.sdk.operatorframework.io][plugin-manifest]: perform the required scaffolds to provide the helpers to allow the projects to be integrated with OLM. 
+- [scorecard.sdk.operatorframework.io][plugin-scorecard]: perform the required scaffolds to provide the [Scorecard][scorecard] feature.
+
+### Optional/custom plugins
+
+Users can also use custom plugins when they execute the SDK CLI sub-commands as a helper for its options which could be like:
```
Comment (marc-obrien) [inline website/content/en/docs/contribution-guidelines/plugins.md:46]: Users can also use custom plugins when they execute the SDK CLI sub-commands as a helper which includes the following options:

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+```sh
+operator-sdk create api --plugins="go/v3,declarative"
+```
+
+The above example will scaffold a custom code in the controllers after an API is created to allow the users to develop solutions using the [kubebuilder declarative pattern][kubebuilder-declarative-pattern]. (e.g [default scaffold][default-scaffold] versus [example][kubebuilder-declarative-pattern-example]).
```
Comment (marc-obrien) [inline website/content/en/docs/contribution-guidelines/plugins.md:52]: The above example will scaffold custom code in the controllers after an API is created to allow the users to develop solutions using the [kubebuilder declarative pattern][kubebuilder-declarative-pattern]. (e.g [default scaffold][default-scaffold] versus [example][kubebuilder-declarative-pattern-example]).

**Review by marc-obrien — CHANGES_REQUESTED** [review]

File: website/content/en/docs/contribution-guidelines/plugins.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+Note that custom plugins are called via the init sub-command to work as global plugins, and will be added in the `layout` field of the PROJECT file. Any sub-command executed will then also be called.
+
+## Plugins Vision
+
+Maintainers and contributors are able to create their own plugin(s) using the same standards and approaches. Theoretically, that could be an option to allow these solutions to be compatible and useful with SDK CLI in the future or at least could result in lower effort to add them on this project, if that is its ultimate motivation. Currently, SDK cannot discover and use plugins that are not shipped or built with its binary. However, it has been discussed in the issue [kubebuilder/issues/1378][kb-issue].
+
+## How to create your own plugins
+
+If you are looking to develop similar solutions to allow users, for example, to create projects using other languages or that could work as helpers for the projects built with SDK/Kubebuilder projects, then see the [Creating your own plugins][create-your-own-plugins] to know how you can do that following this approach as its benefits.
```
Comment (marc-obrien) [inline website/content/en/docs/contribution-guidelines/plugins.md:62]: If you are looking to develop similar solutions to allow users for example to create projects using other languages or that could work as helpers for the projects built with SDK/Kubebuilder projects, then see the [Creating your own plugins][create-your-own-plugins] to see how you can benefit and apply this approach.

**Review by marc-obrien — CHANGES_REQUESTED** [review]

**Review by marc-obrien — APPROVED** [review]
/lgtm

**Pull request merged**